File: MyGroupAPI/Models/UserPay.cs
using System;
using System.Collections.Generic;

namespace MyGroupAPI.Models {
    public class UserPay {
        public int UserPayId { get; set; }
        public DateTime PayDate { get; set; }
        public string AttendSituation { get; set; }
        public double Price { get; set; }
        public double Discount { get; set; }
        public double PriceWithDescount { get; set; }

        public User User { get; set; }
        public int UserId { get; set; }
        
        public UserPay () {
            this.PayDate = DateTime.Now;
            this.PriceWithDescount = this.Price - this.PriceWithDescount;

        }
    }
}

File: MyGroupSPA/src/app/_models/user-pay.ts
import { User } from './user';

export interface UserPay {
    userPayId?:number;
    payDate?:Date;
    attendSituation?:string;
    price?:number;
    discount?:number;
    priceWithDescount?:number;
    user?:User;
    userId?:number;
    arabicName?:string;
    guardianName?:string;
    userGroupName?:string;
    userClassName?:string;
}
